### PR TITLE
Refactor recvPacket tests to use expError.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Using `expError` since `expResult` didn't make as much sense here considering we don't have as much post state to check for, diffs are a mess.

closes: #4448 

### Commit Message / Changelog Entry

```bash
chore: refactor RecvPacket tests to check for the expected error.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/packet_test.go [hunk 1]
```diff
@@ -611,11 +656,7 @@ func (suite *KeeperTestSuite) TestRecvPacket() {
 				}
 			} else {
 				suite.Require().Error(err)
-
-				// only check if expError is set, since not all error codes can be known
-				if expError != nil {
-					suite.Require().True(errors.Is(err, expError))
-				}
+				suite.Require().True(errors.Is(err, tc.expError))
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/packet_test.go:659]: nit: I think you can do: `suite.Require().ErrorIs(err, tc.expError)`

**Review by damiannolan — COMMENTED** [review]
Just dropping a suggestion on error assertion. Tests look great, I'll continue to review the rest of them later and can drop approval then! :)

File: modules/core/04-channel/keeper/packet_test.go (hunk 1 above)
Comment (DimitrisJim) [reply modules/core/04-channel/keeper/packet_test.go:659]: ugh, yea, that slipped through

**Review by damiannolan — APPROVED** [review]
Awesome job! We can probably also cherry-pick the merge commit to main (minus the chan upgrades testcases) like I did with `acknowledgePacket` test refactors.

**Pull request merged**